x/auctions/types/params.go

```diff
@@ -0,0 +1,35 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package types
+
+import "fmt"
+
+var ParamsKey = []byte("Params")
+
+func NewParams(
+	auctionEnabled bool,
+) Params {
+	return Params{
+		EnableAuction: auctionEnabled,
+	}
+}
+
+func DefaultParams() Params {
+	return Params{
+		EnableAuction: true,
+	}
+}
+
+func validateBool(i interface{}) error {
+	_, ok := i.(bool)
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", i)
+	}
+
+	return nil
+}
```

Comment:
Is this needed? EnableAuction is already a tiped bool

Reply:
We have checks for this everywhere else we have typed bools so I suppose there is a chance to bypass the type with a raw transaction ?